src/DebugEngineHost.Stub/DebugEngineHost.ref.cs

```diff
@@ -430,6 +430,24 @@ public static void WriteLaunchError(string outputMessage)
         {
             throw new NotImplementedException();
         }
+
+        /// <summary>
+        /// Send command to VS Code to run the commandArgs in its version of a terminal window
+        /// </summary>
+        /// <param name="title">Title for the terminal window</param>
+        /// <param name="workingDirectory">Working directory to start in the terminal window</param>
+        /// <param name="useExternalConsole">True: External Console, False: Integrated Terminal</param>
+        /// <param name="commandArgs">the list of commands, in order, to run.</param>
+        /// <param name="environmentVariables">Environment Variables</param>
+        /// <param name="success">Success callback. In some cases, id contains the pid of the process that was started.</param>
+        /// <param name="failure">Failure callback. Contains the protocol exception that was thrown.</param>
+        /// <returns>true if the message is sent, false if not.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "cwd")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1006:DoNotNestGenericTypesInMemberSignatures")]
+        public static bool TryRunInTerminal(string title, string workingDirectory, bool useExternalConsole, IReadOnlyList<string> commandArgs, IReadOnlyDictionary<string, string> environmentVariables, Action<int?> success, Action<string> failure)
```

Comment:
>bool [](start = 22, length = 4)

I don't think you want a `bool` return value here, because you need to know if this is available up front. I would suggest changing this to `void RunInTerminal` (or better yet `Task<int?> RunInTerminal` and you can get rid of the completion style API :)) and another API for deciding if the host supports a terminal request.